### PR TITLE
Eng 2493/upstream config cli

### PR DESCRIPTION
**Pull request by Leeoc**
Adds upstream configuration functionality.

### CODE REVIEW

File: cloudsmith_cli/cli/commands/upstream.py [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+from ..utils import (
+    fmt_bool,
+    fmt_datetime,
+    maybe_spinner,
+    maybe_truncate_list,
+    maybe_truncate_string,
+)
+from .main import main
+
+UPSTREAM_FORMATS = [
```
Comment (MrTam) [inline cloudsmith_cli/cli/commands/upstream.py:22]: NAB: it's a shame we can't leverage the existing formats API to get a list of supported upstream formats. Perhaps enriching it with an attribute for `supports_upstream`, or query could be neat. This approach does work of course, but ties upstream format releases to a new CLI release.

For example, here we filter packages that use distros: https://github.com/cloudsmith-io/cloudsmith-cli/blob/5933eb1bf09f1d322d51f0112aae223e37116f1c/cloudsmith_cli/core/api/packages.py#L292-L294

File: cloudsmith_cli/cli/commands/upstream.py [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+        "Extra Value 2",
+        "Active",
+        "Mode",
+        "Priority",
+        "Slug Perm",
+        "Updated At",
+        "Verify SSL",
+        "Component",
+        "Distribution Versions",
+        "Upstream Distribution",
```
Comment (MrTam) [inline cloudsmith_cli/cli/commands/upstream.py:58]: Similar to the above, reckon we could hide these for formats that don't support distros?

**Review by MrTam — APPROVED** [review]
LGTM, some suggestions to improve the 'dynamicity' (i don't think that's a word) of the format support and availability of parameters.

File: cloudsmith_cli/cli/commands/upstream.py (hunk 1 above)
Comment (chrisimcevoy) [reply cloudsmith_cli/cli/commands/upstream.py:22]: True. Sounds like a ticket for QOL pile.

File: cloudsmith_cli/cli/commands/upstream.py (hunk 2 above)
Comment (chrisimcevoy) [reply cloudsmith_cli/cli/commands/upstream.py:58]: Will do, cheers.

**Comment by chrisimcevoy** [comment]
I was unable to merge as there were two commits from @Leeoc which were unsigned. I re-committed everything, should merge now.

**Pull request merged**